scipy/linalg/_basic.py

```diff
@@ -243,6 +276,20 @@ def solve(a, b, lower=False, overwrite_a=False,
                                  overwrite_b=overwrite_b)
         _solve_check(n, info)
         rcond, info = sycon(lu, ipvt, anorm)
+    # Diagonal case
+    elif assume_a == 'dia':
+        x = (b1.T / np.diag(a1)).T
```

Comment:
This needs condition number check for ill-conditioning.

Reply:
OK, so minimum over maximum value in this case?